src/Logic/GhostCMSHelper.php

```diff
@@ -0,0 +1,775 @@
+
+			return $ga;
+		
+		}
+
+		// Check for WP user with admin access.
+		$user_query = new \WP_User_Query(
+			array( 
+				'login'    => $user_login,
+				'role__in' => array( 'Administrator', 'Editor', 'Author', 'Contributor' ),
+			)
+		);
+
+		foreach ( $user_query->get_results() as $wp_user ) {
+
+			$this->log( 'Found existing WP User.' );
+
+			// Save old slug for possible redirect.
+			update_user_meta( $wp_user->ID, 'newspack_ghostcms_slug', $json_author_user->slug );
+
+			// Return the first user found.
+			return $wp_user;
+
+		}
+
+		// Create a GA.
+		$ga_id = $this->coauthorsplus_helper->create_guest_author( array( 'display_name' => $json_author_user->name ) );
+
+		if ( is_wp_error( $ga_id ) || ! is_numeric( $ga_id ) || ! ( $ga_id > 0 ) ) {
+
+			$this->log( 'GA create failed: ' . $json_author_user->name, LogLevel::WARNING );
+
+			return 0;
+
+		}
+
+		$this->log( 'Created new GA.' );
+
+		$ga = $this->coauthorsplus_helper->get_guest_author_by_id( $ga_id );
+	
+		// Save old slug for possible redirect.
+		update_post_meta( $ga->ID, 'newspack_ghostcms_slug', $json_author_user->slug );
+
+		return $ga;
+	}
+
+	/**
+	 * Insert JSON tag as category
+	 *
+	 * @param object $json_tag json tag object.
+	 * @return 0|int
+	 */
+	private function insert_json_tag_as_category( object $json_tag ): int {
+
+		// Must have visibility property with value of 'public'.
+		if ( empty( $json_tag->visibility ) || 'public' != $json_tag->visibility ) {
+			
+			$this->log( 'JSON tag not visible. Could not be inserted.', LogLevel::WARNING );
+
+			return 0;
+
+		} 
+		
+		// Check if category exists in db.
+		// Logic from https://github.com/WordPress/wordpress-importer/blob/71bdd41a2aa2c6a0967995ee48021037b39a1097/src/class-wp-import.php#L784-L801 .
+		// phpcs:ignore WordPressVIPMinimum.Functions.RestrictedFunctions.term_exists_term_exists
+		$term_arr = term_exists( $json_tag->slug, 'category' );
+
+		// Category does not exist.
+		if ( ! $term_arr ) {
+
+			// Insert it.
+			$term_arr = wp_insert_term( $json_tag->name, 'category', array( 'slug' => $json_tag->slug ) );
+
+			// Log and return 0 if insert failed.
+			if ( is_wp_error( $term_arr ) ) {
+				$this->log( 'Insert term failed (' . $json_tag->slug . ') ' . $term_arr->get_error_message(), LogLevel::WARNING );
+				return 0;
+			}
+
+			$this->log( 'Inserted category term: ' . $term_arr['term_id'] );
+
+		}
+		
+		// Save old slug for possible redirect.
+		update_term_meta( $term_arr['term_id'], 'newspack_ghostcms_slug', $json_tag->slug );
+
+		return $term_arr['term_id'];
+	}
+
+	/**
+	 * Log wrapper function incase logging needs to be updated in future it can be changed here.
+	 *
+	 * @param string  $message The message to log.
+	 * @param string  $level Info, error, warning, etc.
+	 * @param boolean $exit_on_error For error messages if desired.
+	 * @return void
+	 */
+	private function log( string $message, string $level = 'info', bool $exit_on_error = false ): void {
```

Comment:
This function should not be necessary anymore. Logging supports multilogging and other tricks now: https://github.com/Automattic/newspack-migration-tools/blob/trunk/docs/logging.md